### PR TITLE
Add linting

### PR DESCRIPTION
**Pull request by jnfeinstein**
I've added `org.jmailen.kotlinter` plugin, which is my go-to `ktlint` plugin for `gradle` (`org.jlleitschuh.gradle.ktlint` uses `gradle` features that are being deprecated in 7.0).  The only manual changes are found in the root `build.gradle.kts` and `.editorconfig`.  The rest of the changes are autogenerated using the linter.

The linter is executed using `./gradlew lintKotlin` to check styling, and `./gradlew formatKotlin` to auto-correct styling.

### CODE REVIEW

File: exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/EntityClass.kt [hunk 1]
```diff
@@ -255,50 +256,50 @@ abstract class EntityClass<ID : Comparable<ID>, out T: Entity<ID>>(val table: Id
...
     @JvmName("backReferencedOnOpt")
-    infix fun <TargetID: Comparable<TargetID>, Target: Entity<TargetID>, REF:Comparable<REF>> EntityClass<TargetID, Target>.backReferencedOn(column: Column<REF?>)
-            : ReadOnlyProperty<Entity<ID>, Target> = registerRefRule(column) { BackReference(column, this) }
+    infix fun <TargetID : Comparable<TargetID>, Target : Entity<TargetID>, REF : Comparable<REF>> EntityClass<TargetID, Target>.backReferencedOn(column: Column<REF?>):
+        ReadOnlyProperty<Entity<ID>, Target> = registerRefRule(column) { BackReference(column, this) }
 
-    infix fun <TargetID: Comparable<TargetID>, Target: Entity<TargetID>, REF:Comparable<REF>> EntityClass<TargetID, Target>.optionalBackReferencedOn(column: Column<REF>)
-            = registerRefRule(column) { OptionalBackReference<TargetID, Target, ID, Entity<ID>, REF>(column as Column<REF?>, this) }
+    infix fun <TargetID : Comparable<TargetID>, Target : Entity<TargetID>, REF : Comparable<REF>> EntityClass<TargetID, Target>.optionalBackReferencedOn(column: Column<REF>) =
+        registerRefRule(column) { OptionalBackReference<TargetID, Target, ID, Entity<ID>, REF>(column as Column<REF?>, this) }
```
Comment (jnfeinstein) [inline exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/EntityClass.kt:278]: @Tapac these were the only changes I had to make manually, the linter did not like moving the `=` sign to the previous line.  Ideally in this case I would use brackets and `return`, but the list of generics is long and it is useful to have them computed transparently in this case.

**Comment by jnfeinstein** [comment]
@Tapac just updated this branch.

**Comment by hfhbd** [comment]
How about adding this to the CI?

**Comment by jnfeinstein** [comment]
@hfhbd 99% certain that the plugin adds `lintKotlin` to the test or check phase.  If not then I am not opposed to adding in a separate PR.  I figured this one would install the plugin and baseline the code, which is a process unto itself.

**Comment by Tapac** [comment]
Your PR was squashed to master manually, so I'll close the PR.

**Comment by BorzdeG** [comment]
Why did you use kotliner? why not https://github.com/arturbosch/detekt/?

**Comment by jnfeinstein** [comment]
Kotlinter is a gradle plugin wrapper for ktlint.  I've used it before, it's easy and works well.  Please submit a PR if you think the project could benefit from using detekt. 🙂

**Pull request closed without merging**